ocaml/xapi/xapi.ml

```diff
@@ -516,16 +478,57 @@ let server_run_in_emergency_mode () =
   wait_to_die () ; exit 0
 
 let update_certificates ~__context () =
+  info "syncing certificates on xapi start" ;
   match Certificates_sync.update ~__context with
   | Ok () ->
-      ()
+      info "successfully synced certificates"
   | Error (`Msg (msg, _)) ->
       error "Failed to update host certificates: %s" msg ;
       server_run_in_emergency_mode ()
   | exception e ->
       error "Failed to update host certificates: %s" (Printexc.to_string e) ;
       server_run_in_emergency_mode ()
 
+let bring_up_management_if ~__context () =
+  update_certificates ~__context |> Xapi_mgmt_iface.add_stunnel_cb ;
+  try
+    let management_if =
+      Xapi_inventory.lookup Xapi_inventory._management_interface
+    in
+    let management_address_type =
+      Record_util.primary_address_type_of_string
+        (Xapi_inventory.lookup Xapi_inventory._management_address_type)
+    in
+    if management_if = "" then (
+      (* we enter this branch during first boot *)
+      debug "No management interface defined (management is disabled)" ;
+      Xapi_mgmt_iface.run ~__context ~mgmt_enabled:false
+    ) else (
+      Xapi_mgmt_iface.change management_if management_address_type ;
+      Xapi_mgmt_iface.run ~__context ~mgmt_enabled:true ;
+      match Helpers.get_management_ip_addr ~__context with
+      | Some "127.0.0.1" ->
```

Comment:
https://stackoverflow.com/questions/40189084/what-is-ipv6-for-localhost-and-0-0-0-0 could be relevant but I'm not familiar with this. This would mean adding cases for "::1" and "0:0:0:0:0:0:0:1".

Reply:
I don't really want to change this (I only moved this function https://github.com/xapi-project/xen-api/pull/4288/commits/df4afe3ac7fcdcc3e99a8441a6d7dd5303f0e4d3), but indeed it looks like we are missing these cases

Maybe @benjamreis has plans to update this alongside his IPv6 changes

Reply:
I'll take a look.
What's the use case? to have the mngm iface on localhost?

Reply:
I'm not sure why you would ever want to set localhost as the management IP address to be localhost... maybe this is here for testing purposes